### PR TITLE
Mobile Nav Fixes

### LINKED ISSUE #13: Mobile Menu Bugs
I noticed the following bugs when doing mobile QA for NSRDB. All issue below are only on mobile navigation.

1. Not all sub menu items are rendered in view for the user. If submenu is more than 3 items they start to get hidden.  
2. Menu doesn't scroll so if there is overflow a user can't access the hidden items. 
3. Submenu chevron will rotate twice really quickly 
4. Bug where when two submenu's are open. Trying to close one of them causes the submenu to still be rendered behind other menu items. Steps to reproduce: Open NSRDB mobile nav. Expand Data sets submenu. Expand about submenu. Click on data sets submenu to attempt to collapse it. Notice that some submenu items are rendering below contact us item. 
5. Bug on desktop browser smaller than 767px so it uses mobile style menu. Mouse hover does not expand the submenus properly. 
<br>
<br>
1.

Screen Shot 2022-01-24 at 2 10 09 PM

<br>
<br>
3. 

https://user-images.githubusercontent.com/19982054/151038973-1444b6cf-6db9-465c-9528-fb025f631b75.mov

<br>
<br>
4. 

Screen Shot 2022-01-27 at 3 33 58 PM

<br>
<br>
5. 

https://user-images.githubusercontent.com/19982054/151591211-f9217bc7-200e-44dd-aa0d-00a4d201b7f4.mov

### PR DESCRIPTION
**Pull request by kyro-D**
Pr closes #13. 

I included a dependency update from running `npm audix fix` with this dev. Let me know if I should revert that back. 

Changes are mostly css changes. Most notable functional change is that the mobile menu no longer supports hover to expand sub menu. This was done by implementing a mobile specific menu item. The items now have to be clicked to expand and show submenu. Hovering over submenus was clunky and a bad user experience when the submenu is long and drastically changes the overall menu height.

### CODE REVIEW

File: src/components/menu/style.scss [hunk 1]
```diff
@@ -83,23 +83,35 @@
...
+      }
     }
   }
 
   // MOBILE MENU
   @media (max-width: 767px) {
     ul.menu {
-      > li.menu-item.show {
-        height: 192px;
-
```
Comment (kyro-D) [inline src/components/menu/style.scss:94]: Issue 1 in ticket.

File: src/components/header/style.scss [hunk 2]
```diff
@@ -248,11 +248,11 @@ $md-header-width: 720px;
...
         &.open {
           transition: 0.3s;
-          height: 100vh;
+          height: calc(100vh - 43px);
           position: fixed;
-          top: 44px;
+          top: 43px;
           background: #ededed;
-          overflow: hidden;
+          overflow: auto;
```
Comment (kyro-D) [inline src/components/header/style.scss:255]: Issue 2 in ticket fix.

File: src/components/menu/style.scss [hunk 3]
```diff
@@ -83,23 +83,35 @@
...
+      > li.menu-item.show,
+      > li.menu-item.active {
         .sub-menu {
           display: block;
+          position: unset;
         }
 
         .submenu-chevron {
-          transform: rotate(270deg);
+          transform: rotate(90deg);
```
Comment (kyro-D) [inline src/components/menu/style.scss:106]: Issue 3 in ticket fix.

File: src/components/menu/style.scss [hunk 4]
```diff
@@ -83,23 +83,35 @@
...
   @media (max-width: 767px) {
     ul.menu {
-      > li.menu-item.show {
-        height: 192px;
-
+      > li.menu-item.show,
+      > li.menu-item.active {
         .sub-menu {
           display: block;
+          position: unset;
```
Comment (kyro-D) [inline src/components/menu/style.scss:102]: Also part of issue 1 fix.

File: src/components/menu/sub-menu.tsx [hunk 5]
```diff
@@ -68,7 +68,16 @@ function SubMenu({
...
+  const toggleShowItems = () => {
+    // avoid edge case of showItems = false and showSubMenu = true
+    if(showSubMenu && showItems){
+      setShowSubMenu(false);
+      setShowItems(false);
+    }
+    else{
+      setShowItems(!showItems);
+    }
+  }
```
Comment (kyro-D) [inline src/components/menu/sub-menu.tsx:80]: Issue 4 in ticket fix.

File: src/components/menu/sub-menu.tsx [hunk 6]
```diff
@@ -97,26 +106,50 @@ function SubMenu({
...
+        </span>
+        <Menu isSubMenu>
+          {/* Need to pass through toggleMenu so that the menu closes when a MenuLink is clicked */}
+          {Children.map(children, child => (
+            cloneElement(child as ReactElement, { toggleMenu })
+          ))}
+        </Menu>
+      </li>
+
+      {/* mobile submenu */}
```
Comment (kyro-D) [inline src/components/menu/sub-menu.tsx:132]: issue 5 in ticket fix. Mobile submenu doesn't have the onMouseEnter and onMouseLeave to remove hovering functionality.

File: src/components/menu/sub-menu.tsx [hunk 7]
```diff
@@ -97,26 +106,49 @@ function SubMenu({
...
+          <ChevronLeft className="submenu-chevron" />
+        </span>
+        <Menu isSubMenu>
+          {/* Need to pass through toggleMenu so that the menu closes when a MenuLink is clicked */}
+          {Children.map(children, child => (
+            cloneElement(child as ReactElement, { toggleMenu })
+          ))}
+        </Menu>
+      </li>
+    </>
```
Comment (kyro-D) [inline src/components/menu/sub-menu.tsx:151]: Issue 5 in ticket fix. Mobile submenu doesn't have the onMouseEnter and onMouseLeave to remove hovering functionality.

**Review by psusmars — CHANGES_REQUESTED** [review]
Everything looks great, but I might just modify `handleShowSubMenu` to check if it's coming through hover and decide on screensize whether or not to run the event handler. Having 2 separate DOM flows seems like a bit.

File: src/components/menu/sub-menu.tsx (hunk 7 above)
Comment (psusmars) [reply src/components/menu/sub-menu.tsx:151]: I think it's a bit excessive to render 2 completely separate dom flows for mobile vs. non-mobile. I might just check the screen size in the event handler of mouse enter to decide if the event should be triggered or not.

**Comment by kyro-D** [comment]
> Everything looks great, but I might just modify `handleShowSubMenu` to check if it's coming through hover and decide on screensize whether or not to run the event handler. Having 2 separate DOM flows seems like a bit.

@psusmars Removed the multiple DOM flows and now check for the window width and do no op if on mobile menu size.

**Review by psusmars — APPROVED** [review]

**Pull request merged**